src/twisted/internet/base.py

```diff
@@ -316,6 +360,16 @@ def getHostByName(self, name, timeout=(1, 3, 11, 45)):
             return defer.succeed(address)
 
 
+_ThreePhaseEventTriggerCallable = Callable[..., Any]
+_ThreePhaseEventTrigger = Tuple[
+    _ThreePhaseEventTriggerCallable, Tuple[object, ...], Dict[str, object]
+]
+_ThreePhaseEventTriggerHandle = NewType(
+    "_ThreePhaseEventTriggerHandle",
+    Tuple[str, _ThreePhaseEventTriggerCallable, Tuple[object, ...], Dict[str, object]],
+)
+
+
```

Comment:
Internal types.
I used `NewType` for the handle that is returned to clients to make it more strictly opaque.